### ISSUE TITLE
Deprecate address floor

### ISSUE THREAD

**Issue by alexsegura** [coopcycle/coopcycle-web#1351]
The address floor field makes no real sense.
It's a text field, and people enter pretty much everything in it. 
Even transforming it to a number would not really make sense.

**Comment by PasLoin** [comment]
Transforming it into number can make sense in case of a lot of multiple points in same building ; for optimisation ; but can be done in data source also.

**Comment by joel-shiftdelivery** [comment]
For us, while the name 'floor' doesn't make a lot of sense, we really need a field for the unit number.

Most of our deliveries have a unit number for an apartment (or office) and often a buzzer number for the recipient to let us in.
We currently use the 'floor' field for the unit number (almost always just numbers, with a rare letter ex: unit 36A), with the instructions or notes section for the buzzer number.

Due to the way CoopCycle (and Google) handle addresses, we can not include the unit number in the address. Ex: 3603-1288 W Georgia St, Vancouver, BC 
3603 is the unit and 1288 is the street address.
This address either does not parse, is read incorrectly (3603 W Georgia St instead of 1288 W Georgia St), or it does not save the unit number so it's just 1288 W Georgia St with no unit (which is very unhelpful).

For us it would be helpful to have a number field for unit number, and an optional text field for buzzer/delivery instructions that would become required if someone adds a unit number.


Basically, here is the fields we need:
Address
Unit number
Customer name
Phone
Buzzer or delivery instructions

Name and Contact name seem a bit redundant of each other, but I know there's probably a reason behind it.
Instructions and Comments initially seem similarly redundant, but I can potentially see it being useful if one is used for customer provided info and the other is internal use only.

**Comment by alexsegura** [comment]
@joel-shiftdelivery ok, thanks for the details.

If I understand well, the "unit number" is something that is specific to the address system in Canada, right? In this case, it would be best to have a "localized" behaviour for addresses for Canada only. I mean, this "unit number" would appear only for instances running with country = CA.

The address system is something complex, there's also something specific in the UK (see #1288)

> Name and Contact name seem a bit redundant of each other, but I know there's probably a reason behind it.

Actually it's not clear, but "name" property of address is an "alias", a shortcut. Like "Mom" for the address of your mother 🙂

**Comment by joel-shiftdelivery** [comment]
Hmmmm... Thinking about it more, I think unit numbers are not really part of the address system, but are necessary information to complete a delivery (we have lots of tower apartments where we deliver to the unit door).

I think your solution would work well, so there would be a field for "unit number" for instances in CA.
